SFR/Assembly/inner_assembly/make_SFR_assembly_model.py

```diff
@@ -0,0 +1,209 @@
+
+
+###################################################################
+##########################   Materials     ########################
+###################################################################
+
+u235 = openmc.Material(name='U235')
+u235.add_nuclide('U235', 1.0)
+u235.set_density('g/cm3', 10.0)
+
+u238 = openmc.Material(name='U238')
+u238.add_nuclide('U238', 1.0)
+u238.set_density('g/cm3', 10.0)
+
+pu238 = openmc.Material(name='Pu238')
+pu238.add_nuclide('Pu238', 1.0)
+pu238.set_density('g/cm3', 10.0)
+
+pu239 = openmc.Material(name='U235')
+pu239.add_nuclide('Pu239', 1.0)
+pu239.set_density('g/cm3', 10.0)
+
+pu240 = openmc.Material(name='Pu240')
+pu240.add_nuclide('Pu240', 1.0)
+pu240.set_density('g/cm3', 10.0)
+
+pu241 = openmc.Material(name='Pu241')
+pu241.add_nuclide('Pu241', 1.0)
+pu241.set_density('g/cm3', 10.0)
+
+pu242 = openmc.Material(name='Pu242')
+pu242.add_nuclide('Pu242', 1.0)
+pu242.set_density('g/cm3', 10.0)
+
+am241 = openmc.Material(name='Am241')
+am241.add_nuclide('Am241', 1.0)
+am241.set_density('g/cm3', 10.0)
+
+o16 = openmc.Material(name='O16')
+o16.add_nuclide('O16', 1.0)
+o16.set_density('g/cm3', 10.0)
+
+sodium = openmc.Material(name='Na')
+sodium.add_nuclide('Na23', 1.0)
+sodium.set_density('g/cm3', 0.96)
+
+cu63 = openmc.Material(name='Cu63')
+cu63.set_density('g/cm3', 10.0)
+cu63.add_nuclide('Cu63', 1.0)
+
+Al2O3 = openmc.Material(name='Al2O3')
+Al2O3.set_density('g/cm3', 10.0)
+Al2O3.add_element('O', 3.0)
+Al2O3.add_element('Al', 2.0)
+
+helium = openmc.Material(name='Helium for gap')
+helium.set_density('g/cm3', 0.001598)
+helium.add_element('He', 2.4044e-4)
+
+#Material mixtures
+
+inner_fuel_material= openmc.Material.mix_materials([u235, u238, pu238, pu239, pu240, pu241, pu242, am241, o16],[0.0019, 0.7509, 0.0046, 0.0612, 0.0383, 0.0106, 0.0134, 0.001, 0.1181],'wo')
+cladding_material = openmc.Material.mix_materials([cu63,Al2O3], [0.997,0.003], 'wo')
+
+materials=openmc.Materials([inner_fuel_material,sodium,helium,cladding_material])
+materials.export_to_xml()
+
+
+###############################################################################
+# Define problem geometry
+
+# Create cylindrical surfaces
+
+
+
+# Create cylindrical surfacesfuel_or = openmc.ZCylinder(surface_id=1, r=0.943/2) 
+
+fuel_or = openmc.ZCylinder(surface_id=1, r=0.943/2) 
+clad_ir = openmc.ZCylinder(surface_id=2, r=0.973/2) 
+clad_or = openmc.ZCylinder(surface_id=3, r=1.073/2) 
+
+
+
+z_co_ordinates=np.linspace(-height/2,height/2,N+1)
+z_plane=[]
+for i in range (len(z_co_ordinates)):
+    z_plane.append(openmc.ZPlane(z0=z_co_ordinates[i]))
+
+
+#set BCs
+z_plane[0].boundary_type='vacuum'
+z_plane[-1].boundary_type='vacuum'
+top=z_plane[-1]
+bottom=z_plane[0]
+
+inner_fuel_cells=[]
+inner_gas_gap_cells=[]
+inner_clad_cells=[]
+inner_sodium_cells=[]
+
+all_inner_cells=[]
+
+outer_fuel_cells=[]
+outer_gas_gap_cells=[]
+outer_clad_cells=[]
+outer_sodium_cells=[]
+
+for i in range (N):
+    
+    layer=+z_plane[i]&-z_plane[i+1]
```

Comment:
This may have been from an example, but why not use a lattice in the z-direction? I would expect the tracking performance to be a little bit better that way, as opposed to N unique cells.

Reply:
Agreed. Check out my GCMR [code](https://github.com/cnerg/gcmr/blob/7b7f7785f7aeb0068c279975434c76f0c138196b/depletion_standalone/inf_assemblies/make_gcmr_model.py#L158) for an example with a 3D lattice. There might be some depletion things you don't need in there but it should serve as a good base. IIRC there's some cardinal tutorials that do it too

Reply:
This may be based on the approach that @nuclearkevin  used for LWR problems?